### PR TITLE
Fixes #1423

### LINKED ISSUE #1423: WriteGPIO and ReadGPIO on Pn532 are not implemented as Pn532 specs define
**Describe the bug**

I've been trying to use the WriteGPIO and ReadGPIO methods defined on Pn532 class and wouldn't be able to modify the values of Port7. Doing some research on how WriteGPIO and ReadGPIO are implemented I found 2 issues on the write method and 1 issue on the read method.

WriteGPIO method:
The write gpio message should be sent as the following = [Command_Identifier, Port3_Value, Port7_Value] where:
Command_Identifier = 0x0E (Write Gpio Command)

If we are going to modify values on Port3 then bit 7 on this value should be 1, otherwise bit 7 should be 0. Same logic applies to Port7.

If you take a look at how WriteGPIO method is implemented, this bit modification is never done so bit 7 for any port is set to 0 everytime and no modification on Ports values is done by the board.

In addition the message is being built with the following structure: [Command_Identifier, Port7_Value, Port3_Value] which is not how is defined by the specifications of Pn532.

ReadGPIO method:
Response for Read GPIO Command on Pn532 should be interpreted as the following: [Port3_Value, Port7_Value, Operating_Mode] as it's stated on Pn532 specifications.

Implementation on this method interprets response as the following [Port7_Value, Port3_Value, Operating_Mode]

Pn532 specifications: https://www.nxp.com/docs/en/user-guide/141520.pdf. 
Write Gpio => page 81
Read Gpio => page 79

**Steps to reproduce**

```
var pn = new Pn532("/dev/ttyO4");
pn.SetSerialBaudRate(BaudRate.B0115200);
pn.WakeUp();
            
var readGpio = pn.ReadGpio(out var p7, out var p3, out var operatingMode);
var writeGpio = pn.WriteGpio((Port7)0, (Port3)0);
readGpio = pn.ReadGpio(out p7, out p3, out operatingMode);
```
**Expected behavior**

p7 and p3 values should be 0

**Actual behavior**

p3 and p7 values are not 0

**Versions used**

Add following information:

- `dotnet --info` on the machine being used to build
.NET SDK (reflecting any global.json):
 Version:   5.0.101
 Commit:    d05174dc5a

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.18363
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\5.0.101\

Host (useful for support):
  Version: 5.0.1
  Commit:  b02e13abab

.NET SDKs installed:
  2.1.509 [C:\Program Files\dotnet\sdk]
  2.1.603 [C:\Program Files\dotnet\sdk]
  2.2.203 [C:\Program Files\dotnet\sdk]
  3.0.103 [C:\Program Files\dotnet\sdk]
  5.0.101 [C:\Program Files\dotnet\sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.All 2.1.10 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.1.13 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.1.23 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.2.4 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.1.10 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.1.13 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.1.23 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.2.4 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.0.3 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.10 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 5.0.1 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 2.1.10 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.13 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.23 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.2.4 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.0.3 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.10 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 5.0.1 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 3.0.3 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 3.1.10 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 5.0.1 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

To install additional .NET runtimes or SDKs:
  https://aka.ms/dotnet-download

- Version of `System.Device.Gpio` package 1.3.0
- Version of `Iot.Device.Bindings` package 1.3.0

### LINKED ISSUE #1432: Ssd1306 controller does not work properly after sending first chunk of data
**Describe the bug**

I've been testing the Oled Ssd1306 driver and found that after sending the first chunk of data to be stored on ram and displayed on screen commands do not work properly.

After doing some research on the code I found that the error relies on the SendCommand implementation because it doesn't set the control byte to 0x00 before sending the command's byte array. 

In order to fix this issue one line shoud be added to SendCommand implementation to set this value as 0x00:

```
            Span<byte> writeBuffer = SliceGenericBuffer(commandBytes.Length + 1);
            writeBuffer[0] = 0x00; // Control byte <== FIX
            commandBytes.CopyTo(writeBuffer.Slice(1));

            // Be aware there is a Continuation Bit in the Control byte and can be used
            // to state (logic LOW) if there is only data bytes to follow.
            // This binding separates commands and data by using SendCommand and SendData.
            _i2cDevice.Write(writeBuffer);
```

**Steps to reproduce**

```csharp
var _ssd1306 = new Ssd1306(I2cDevice.Create(new I2cConnectionSettings(bus, deviceAddress)));

// Inititalize

_ssd1306.SendCommand(new SetDisplayOff());
_ssd1306.SendCommand(new SetDisplayClockDivideRatioOscillatorFrequency());
_ssd1306.SendCommand(new SetMultiplexRatio((byte)(_height - 1)));
_ssd1306.SendCommand(new SetDisplayOffset());
_ssd1306.SendCommand(new SetDisplayStartLine());
_ssd1306.SendCommand(new SetChargePump(true));
_ssd1306.SendCommand(new SetMemoryAddressingMode(SetMemoryAddressingMode.AddressingMode.Horizontal));
_ssd1306.SendCommand(new SetSegmentReMap(true));
_ssd1306.SendCommand(new SetComOutputScanDirection(false));
_ssd1306.SendCommand(new SetComPinsHardwareConfiguration());
_ssd1306.SendCommand(new SetContrastControlForBank0(0xCF));
_ssd1306.SendCommand(new SetPreChargePeriod(0x01, 0x0F));
_ssd1306.SendCommand(new SetVcomhDeselectLevel(DeselectLevel.Vcc1_00));
_ssd1306.SendCommand(new EntireDisplayOn(false));
_ssd1306.SendCommand(new SetNormalDisplay());
_ssd1306.SendCommand(new DeactivateScroll());
_ssd1306.SendCommand(new SetDisplayOn());

_ssd1306.SendCommand(new SetColumnAddress(0, (byte)ColEndAddress));
_ssd1306.SendCommand(new SetPageAddress());

_ssd1306.SendData(new byte[]{0x40, 0xFF, 0xFF, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0xFF, 0xFF, 0x00, 0x00, 0xC0, 0xC0, 0x30, 0x30});
_ssd1306.SendData(new byte[]{0x40, 0x30, 0x30, 0x30, 0x30, 0xC0, 0xC0, 0x00, 0x00, 0xF0, 0xF0, 0xC0, 0xC0, 0x30, 0x30, 0x30, 0x30});
_ssd1306.SendData(new byte[]{0x40, 0xC0, 0xC0, 0x00, 0x00, 0xC0, 0xC0, 0x30, 0x30, 0x30, 0x30, 0x30, 0x30, 0xC0, 0xC0, 0x00, 0x00});
_ssd1306.SendData(new byte[]{0x40, 0xF0, 0xF0, 0x30, 0x30, 0xC0, 0xC0, 0x30, 0x30, 0xC0, 0xC0, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00});
_ssd1306.SendData(new byte[]{0x40, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x03, 0x03, 0xFF, 0xFF, 0x03, 0x03});
_ssd1306.SendData(new byte[]{0x40, 0x00, 0x00, 0x00, 0x00, 0xFF, 0xFF, 0x03, 0x03, 0x03, 0x03, 0x03, 0x03, 0xFC, 0xFC, 0x00, 0x00});
_ssd1306.SendData(new byte[]{0x40, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00});
_ssd1306.SendData(new byte[]{0x40, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00});
_ssd1306.SendData(new byte[]{0x40, 0x03, 0x03, 0x0C, 0x0C, 0x30, 0x30, 0x0C, 0x0C, 0x03, 0x03, 0x00, 0x00, 0x0F, 0x0F, 0x33, 0x33});
_ssd1306.SendData(new byte[]{0x40, 0x33, 0x33, 0x33, 0x33, 0x03, 0x03, 0x00, 0x00, 0x3F, 0x3F, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00});
_ssd1306.SendData(new byte[]{0x40, 0x3F, 0x3F, 0x00, 0x00, 0x0F, 0x0F, 0x30, 0x30, 0x30, 0x30, 0x30, 0x30, 0x0F, 0x0F, 0x00, 0x00});
_ssd1306.SendData(new byte[]{0x40, 0x3F, 0x3F, 0x00, 0x00, 0x3F, 0x3F, 0x00, 0x00, 0x3F, 0x3F, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00});
_ssd1306.SendData(new byte[]{0x40, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x30, 0x30, 0x3F, 0x3F, 0x30, 0x30});
_ssd1306.SendData(new byte[]{0x40, 0x00, 0x00, 0x00, 0x00, 0x3F, 0x3F, 0x30, 0x30, 0x30, 0x30, 0x30, 0x30, 0x0F, 0x0F, 0x00, 0x00});
```

**Expected behavior**

After sending data for the first time to the display, it doesn't work as expected.

**Actual behavior**

Continue working properly

**Versions used**

Add following information:

.NET SDK (reflecting any global.json):
 Version:   5.0.101
 Commit:    d05174dc5a

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.18363
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\5.0.101\

Host (useful for support):
  Version: 5.0.1
  Commit:  b02e13abab

.NET SDKs installed:
  2.1.509 [C:\Program Files\dotnet\sdk]
  2.1.603 [C:\Program Files\dotnet\sdk]
  2.2.203 [C:\Program Files\dotnet\sdk]
  3.0.103 [C:\Program Files\dotnet\sdk]
  5.0.101 [C:\Program Files\dotnet\sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.All 2.1.10 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.1.13 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.1.23 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.2.4 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.1.10 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.1.13 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.1.23 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.2.4 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.0.3 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.10 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 5.0.1 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 2.1.10 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.13 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.23 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.2.4 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.0.3 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.10 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 5.0.1 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 3.0.3 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 3.1.10 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 5.0.1 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

### PR DESCRIPTION
**Pull request by rodrigobengoechea**
Fixes #1423 :
Fixing methods WriteGPIO and ReadGPIO on Pn532 device. Now methods are implemented according Pn532 specifications.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/iot/pull/1434)

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/dotnet/iot?pullRequest=1434) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/iot?pullRequest=1434) before we can accept your contribution.<br/><br/>:x: rodrigobengoechea [sign now](https://cla.dotnetfoundation.org/dotnet/iot?pullRequest=1434)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/iot?pullRequest=1434) it.</sub>

**Comment by rodrigobengoechea** [comment]
Title is wrong due to a typo. Bug fixed is #1423.

File: src/devices/Pn532/samples/Program.cs [hunk 1]
```diff
@@ -53,7 +54,7 @@
 }
 else if (choiceInterface is { KeyChar: '2' })
 {
-    pn532 = new Pn532(I2cDevice.Create(new I2cConnectionSettings(1, Pn532.I2cDefaultAddress)), debugLevel);
+    pn532 = new Pn532(I2cDevice.Create(new I2cConnectionSettings(2, Pn532.I2cDefaultAddress)), debugLevel);
```
Comment (Ellerbach) [inline src/devices/Pn532/samples/Program.cs:57]: Why this change from 1 to 2?

File: src/devices/Pn532/samples/Program.cs [hunk 2]
```diff
@@ -75,7 +76,8 @@
 
     // To run tests, uncomment the next line
     // RunTests(pn532);
-    ReadMiFare(pn532);
+    // ReadMiFare(pn532);
```
Comment (Ellerbach) [inline src/devices/Pn532/samples/Program.cs:79]: Can you uncomment the Mifare part and comment the GPIO one? I guess most people are interested in reading the Mifare cards.

**Review by Ellerbach — APPROVED** [review]
Thanks for providing the fix! Really appreciate. Just a little question and a comment. then we'll be good to merge.
Note: just commit on your own branch and push on your own repo, everything will automatically appear here

File: src/devices/Pn532/samples/Program.cs (hunk 1 above)
Comment (rodrigobengoechea) [reply src/devices/Pn532/samples/Program.cs:57]: I've changed becuase I was connecting P532 to bus 2, will undo that change

File: src/devices/Pn532/samples/Program.cs (hunk 2 above)
Comment (rodrigobengoechea) [reply src/devices/Pn532/samples/Program.cs:79]: sure

**Review by krwq — APPROVED** [review]

**Comment by Ellerbach** [comment]
Thanks again @rodrigobengoechea for providing the fix as a PR! I hope you'll enjoy even more .NET IoT :-D

**Comment by rodrigobengoechea** [comment]
Thanks to you! Is there any estimate date when these fixes will be a available on a new package versión?

**Comment by Ellerbach** [comment]
> Is there any estimate date when these fixes will be a available on a new package versión?

1.4 is in preparation, should be matter of days/a week or so. We're trying now to get one release per month approximately.

**Pull request merged**